data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -110,6 +112,48 @@ public GetForkResponse getForkInfo() {
                             .collect(toList())));
   }
 
+  public SafeFuture<Optional<List<EpochCommitteeResponse>>> getCommitteesAtEpochV1(
+      final UInt64 stateSlot,
+      final UInt64 epoch,
+      final Optional<UInt64> maybeSlot,
+      final Optional<Integer> maybeIndex) {
+    if (!combinedChainDataClient.isChainDataFullyAvailable()) {
+      return chainUnavailable();
+    }
+
+    final UInt64 earliestQueryableSlot =
+        CommitteeUtil.getEarliestQueryableSlotForTargetEpoch(epoch);
+    if (recentChainData.getHeadSlot().isLessThan(earliestQueryableSlot)) {
+      return SafeFuture.completedFuture(Optional.empty());
+    }
+
+    if (stateSlot.isLessThan(earliestQueryableSlot)) {
+      return SafeFuture.completedFuture(Optional.empty());
+    }
+
+    return combinedChainDataClient
+        .getStateAtSlotExact(stateSlot)
+        .thenApply(
+            maybeResult ->
+                maybeResult.map(
+                    state ->
+                        combinedChainDataClient.getCommitteesFromState(state, epoch).stream()
+                            .filter(
+                                committee ->
+                                    maybeSlot
+                                        .map(slot -> committee.getSlot().equals(slot))
+                                        .orElse(true))
+                            .filter(
+                                committee ->
+                                    maybeIndex
+                                        .map(
+                                            index ->
+                                                committee.getCommitteeIndex().intValue() == index)
+                                        .orElse(true))
+                            .map(EpochCommitteeResponse::new)
+                            .collect(toList())));
```

Comment:
possible to make that a bit easier to digest?

Reply:
Donne.